engine/src/rocksdb_engine.rs

```diff
@@ -0,0 +1,201 @@
+use std::{iter::repeat, path::Path, sync::Arc};
+
+use rocksdb::{Error as RocksError, Options, WriteBatchWithTransaction, WriteOptions, DB};
+
+use crate::{
+    engine_api::{StorageEngine, WriteOperation},
+    error::EngineError,
+};
+
+/// Translate a `RocksError` into a `EngineError`
+fn translate_error(err: RocksError) -> EngineError {
```

Comment:
use `From` trait, so that we don't have to `map_err` every time, just use the `?` directly

Reply:
Fixed